micro_sam/sam_annotator/_widgets.py

```diff
@@ -0,0 +1,56 @@
+from enum import Enum
+from pathlib import Path
+from typing import TYPE_CHECKING, Optional
+
+from magicgui import magic_factory
+from magicgui.tqdm import tqdm
+from superqt.utils import thread_worker
+
+from micro_sam.util import (
+    ImageEmbeddings,
+    get_sam_model,
+    precompute_image_embeddings,
+    _MODEL_URLS,
+    _DEFAULT_MODEL,
+)
+
+if TYPE_CHECKING:
+    import napari
+
+Model = Enum("Model", _MODEL_URLS)
+
+
+@magic_factory(call_button="Compute image embeddings",
+               device = {"choices": ["auto", 'cuda', 'mps', 'cpu']})
+def embedding_widget(
+    image: "napari.layers.Image",
+    model: Model = Model.__getitem__(_DEFAULT_MODEL),
+    device = "auto",
+    save_path: Optional[Path] = None,  # where embeddings for this image are cached (optional)
+    custom_model: Optional[str] = None,  # A filepath or URL to custom model weights.
+) -> ImageEmbeddings:
+    """Image embedding widget."""
+    # for access to the predictor and the image embeddings in the widgets
+    global PREDICTOR, IMAGE_EMBEDDINGS
+    # Initialize the model
+    PREDICTOR = get_sam_model(model_type=model.name)
+
+    # Get image dimensions
+    if not image.rgb:
+        ndim = image.data.ndim
+    else:
+        # assumes RGB channels are the last dimension
+        ndim = image.data.ndim
+
+    # Compute the embeddings for the image data
+    with tqdm() as pbar:
+        @thread_worker(connect={"finished": lambda: pbar.progressbar.hide()})
+        def _compute_image_embedding(PREDICTOR, image, save_path, ndim=None):
+            IMAGE_EMBEDDINGS = precompute_image_embeddings(
```

Comment:
I am not completely confident that this saves the image embeddings to the global variable.
I don't work with global variables much, perhaps someone else can help me check this?

Reply:
Specifically, it's the fact this is happening in a different thread that I'm worried about.

Reply:
I am also not so sure about this. I would for sure add 
```python
global IMAGE_EMBEDDINGS
```
below the function signature, from my understanding of global variables that should ensure that the global variable is updated.

Reply:
Ok, I'm going to move `global IMAGE_EMBEDDINGS` out from the `embedding_widget` function, and inside the `_compute_image_embedding` thread worker function. I think it must make more sense to have it in one place, not two. I'll see if there's a way I can check if this is working well.